cmd/controller-manager/app/controller_manager.go

```diff
@@ -90,19 +90,20 @@ func Run(controllerManagerOptions *options.ControllerManagerServer) error {
 	// Build the K8s kubeconfig / client / clientBuilder
 	glog.V(4).Info("Building k8s kubeconfig")
 
-	k8sKubeconfig, err := rest.InClusterConfig()
+	var err error
+	var k8sKubeconfig *rest.Config
+	if controllerManagerOptions.K8sAPIServerURL == "" && controllerManagerOptions.K8sKubeconfigPath == "" {
+		k8sKubeconfig, err = rest.InClusterConfig()
+	} else {
+		k8sKubeconfig, err = clientcmd.BuildConfigFromFlags(
+			controllerManagerOptions.K8sAPIServerURL,
+			controllerManagerOptions.K8sKubeconfigPath)
+	}
 	if err != nil {
-		glog.Fatalf("Failed to get kube client config (%s)", err)
+		return fmt.Errorf("failed to get Kubernetes client config: %v", err)
```

Comment:
it's in the coding conventions for Kubernetes somewhere.  Bascially:
1. Always use lowercase starts, because the error could be nested in something else.  For example, if the error is logged elsewhere, we might have `glog.Errorf("Failed to initialize controller manager: %v", err)`, which yields a sensible `Failed to initialize controller: manager: failed to get Kubernetes client config: couldn't frob the bits`, but if we use uppercase, we get `Failed to initialize the controller manager: Failed to get...`, which looks weird with the extra upper case in the middle of the sentence.

2. errors are structured as `message: %v`.  This makes them chain without nested parentheses.